### ISSUE TITLE
TinyMCE editor tab panels lack ARIA attributes

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/pressbooks#1635]
### Description

General guidance is that all tabpanel functionality should be marked up with ARIA to describe the purpose and interaction of the tab elements. Some tabpanels and tablists lack relevant ARIA attributes. For example, the tabs in TinyMCE that allow users to switch between the visual/text editor currently lack relevant attributes. I'm unsure if there are other tabs more directly under our control that are similarly affected.

**Suggestion:** Add role=”tablist” to the tab `<ul>` and `role=”tab”` to individual `<li>`s. On the tab content divs, add `role=”tabpanel”` and `aria-labeledby=id` of its tab. Set `aria-selected=”true”` for selected tabs, and `aria-hidden=”true”` for hidden tab panels. For an example, see:  http://accessibility.athena-ict.com/aria/examples/tabpanel2.shtml

In some cases, invalid use of ARIA markup was observed as well. Example: aria-level is an invalid attribute on `role=”button”`

![Screen Shot 2019-04-04 at 2 33 33 PM](https://user-images.githubusercontent.com/39064028/55579593-a14aa000-56e6-11e9-9012-7da5f513dff9.png)

**Comment by arzola** [comment]
This issue has been automatically closed as part of an organization-wide backlog review.

**Reason:** This issue was created 6y 11m ago with no updates in the last 5m.

If this is still relevant, please feel free to reopen it with updated context. We appreciate your contributions and want to ensure our issue tracker stays actionable.

**Learn more:** https://pressbooks.community/t/announcement-automatic-closure-of-old-github-tickets/4542

**Comment by greatislander** [comment]
It's not possible to modify the embedded TinyMCE editor to add the appropriate tab roles. If the plan is to eventually replace TinyMCE with a different editor, I'd suggest this be closed as not planned. @MichSpieler @arzola.
